providers/cookbook_ci.rb

```diff
@@ -0,0 +1,62 @@
+use_inline_resources if defined?(use_inline_resources)
+
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+#
+# create action
+#
+action :create do
+  job_name = "cookbook-#{new_resource.name}"
+
+  job_config = ::File.join(Chef::Config[:file_cache_path], "#{job_name}-config.xml")
+
+  jenkins_job job_name do
+    action :nothing
+    config job_config
+  end
+
+  commands = ["bundle install --deployment"]
+  commands << "bundle exec rake test:foodcritic" if new_resource.foodcritic
+  commands << "bundle exec rake test:chefspec" if new_resource.chefspec
+  commands << "bundle exec berks" if new_resource.serverspec
```

Comment:
I have found not all authors check in their lock files for either berks or gem files. At the moment, this confirms the cookbooks are downloaded into the workspace. I was planning on making the commands more customizable based on the user's whim.